docs/source/project_documentation/aac_gen_gherkin.md

```diff
@@ -0,0 +1,102 @@
+# AaC Gen-Gherkin
+
+The AaC Gen-Gherkin plugin generates gherkin feature files based on provided AaC models.
+
+## gen-gherkin-behaviors Command
+
+```bash
+aac gen-gherkin-behaviors architecture-file.aac output/directory
+```
+
+### Arguments
+
+#### Architecture File
+The AaC file containing the `model` definition/s.
+
+#### Output Directory
+The directory in which the gherkin feature files will be generated.
+
+## Definition Type References
+The `gen-gherkin` plugin will generate feature files based on the `behavior` field of `model` definitions.
+`behavior` definitions also have a field named `acceptance`.  The  `acceptance` field is where the feature files scenario steps will come from.
+
+### Example Model Definition
+
+```yaml
+model:
+    name: example_name
+    description: Description of Model
+    behavior:
+        - name: example_behavior
+          description: Does something
+          acceptance:
+            - name: example_acceptance
+              scenarios:
+                - name: example scenario
+                  given:
+                    - Initial state
+                  when:
+                    - Something happens
+                  then:
+                    - The model does something in response.
+```
+
+### Gherkin Syntax
+When output to a feature file, a `model`s `behavior` becomes a `feature`, and its `acceptance` becomes a `scenario`.
+
+Each `scenario` has a number of steps:
+
+`given`: An optional step, but is usually recommended to include.  The `given` step describes the models initial state.
+
+`when`: This step describes what kicks off the action the `model` will perform, be it called through a command line or triggered some other way.
+
+`then`: This step describes what happens after the `model` performs its action.  Typically it will include whatever output is created by the `model`'s action.
+
+Along with the `scenario` steps, `features` and `scenario`s can also have requirements, signified by the `requirement ID` before the `feature` or `scenario`, such as the following excerpt taken from the `Math Message Handler` feature file:
+```text
+@SQRT-1
+Feature:
+
+    @SQRT-2
+    Scenario: Request handler issues error for square root command with negative value
+
+        Given The MathMessageHandler is running
+        When The user makes a square root request with MathRequest
+        Then The user gets a MathResponse response with the error
+```
```

Comment:
Remember for explicit examples, we use the includes to directly reference an example file and portion so that we don't have to manually update these if we change something with the code or generation, it would just reflect the change on its own.

Reply:
But this is referencing a generated outputted file of the command which doesn't get maintained with the code base...

So, I would include the screen cap of the `mathmessagehandler_api_feature_file.feature` here as well instead.

Reply:
I still think it would be clearer to have this section just be a requirement above a scenario like it was initially:
@SQRT-2
Scenario:


Then the full example is shown later in the document.  This part of the doc is just describing what a requirement is and where it goes, the full context of an entire file makes it less clear I think.

Reply:
Could possibly have a version of the screenshot which highlights the requirements instead, that might make it come across clearer.

Reply:
The modified screenshot would work. Agree that it does make it less clear having a full file provided.